apps/ca.c

```diff
@@ -1240,7 +1240,8 @@ int ca_main(int argc, char **argv)
         if (crl_ext != NULL || crlnumberfile != NULL) {
             X509V3_CTX crlctx;
 
-            X509V3_set_ctx(&crlctx, x509, NULL, NULL, crl, 0);
+            if (!X509V3_set_ctx(&crlctx, x509, NULL, NULL, crl, 0))
+                goto end;
             X509V3_set_nconf(&crlctx, conf);
```

Comment:
Add a check here?

Reply:
All right, added the possibility for `X509V3_set_nconf()` to return error.
Also simplified its use in `x509.c`, `req.c`, and `ca.c`.